storage/innobase/handler/ha_innopart.cc

```diff
@@ -1056,6 +1066,11 @@ ha_innopart::open(
 	m_clust_pcur_parts = NULL;
 	m_pcur_map = NULL;
 
+    if (ib_table->corrupted) {
+        close();
+        DBUG_RETURN(HA_ERR_TABLE_CORRUPT);
+    }
```

Comment:
what does this close()? do :thinking: will it internally call dict_table_close()?

Also, can we do the same for normal tables? ie return HA_ERR_TABLE_CORRUPT on ib_table->corrupted?

Reply:
It iterates over all partitions and closes them.
If the partition is corrupted, dict_table_close() is not called.